lib/block-supports/typography.php

```diff
@@ -92,117 +92,142 @@ function gutenberg_apply_typography_support( $block_type, $block_attributes ) {
 	$has_text_decoration_support = _wp_array_get( $typography_supports, array( '__experimentalTextDecoration' ), false );
 	$has_text_transform_support  = _wp_array_get( $typography_supports, array( '__experimentalTextTransform' ), false );
 
-	if ( $has_font_size_support && ! gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'fontSize' ) ) {
+	// Whether to skip individual block support features.
+	$should_skip_font_size       = gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'fontSize' );
+	$should_skip_font_family     = gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'fontFamily' );
+	$should_skip_font_style      = gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'fontStyle' );
+	$should_skip_font_weight     = gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'fontWeight' );
+	$should_skip_line_height     = gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'lineHeight' );
+	$should_skip_text_decoration = gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'textDecoration' );
+	$should_skip_text_transform  = gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'textTransform' );
+	$should_skip_letter_spacing  = gutenberg_should_skip_block_supports_serialization( $block_type, 'typography', 'letterSpacing' );
+
+	if ( $has_font_size_support && ! $should_skip_font_size ) {
 		$has_named_font_size  = array_key_exists( 'fontSize', $block_attributes );
 		$has_custom_font_size = isset( $block_attributes['style']['typography']['fontSize'] );
 
 		if ( $has_named_font_size ) {
-			$classes[] = sprintf( 'has-%s-font-size', _wp_to_kebab_case( $block_attributes['fontSize'] ) );
+			$classes['fontSize'] = _wp_to_kebab_case( $block_attributes['fontSize'] );
```

Comment:
> convert the value to kebab case to the style engine, too, so that the caller doesn't have to worry about the detail of how the class name is ultimately constructed

It probably should, but would it need to if it's inferring classnames and building them "in house" based on attributes alone? 

The tricky part of the puzzle here is building classnames based on existing patterns. For `typography` only font size received the kebab case treatment. That's the reason why I left it to the block support to pass the right slug, and kept the style engine "dumb" in that regard. 

I can't think of any reason not to do this - in fact I agree that we definitely should ensure that all classnames are consistent in that way - however if ever a block/block support _didn't_ want kebab case (an extreme edge case) and something else, we might have painted ourselves into a corner too early.

Once again, I'm not sure. Maybe we'll know as we add more supports to the mix.

Reply:
> Once again, I'm not sure. Maybe we'll know as we add more supports to the mix.

Good points! I think if we're not sure, then perhaps let's leave it in the block support so that we haven't painted ourselves into a corner, like you say. We could always move it over in a follow-up, if need be.